src/metax_api/api/oaipmh/base/metax_oai_server.py

```diff
@@ -440,6 +442,10 @@ def getRecord(self, metadataPrefix, identifier):
             if metadataPrefix == OAI_DC_URNRESOLVER_MDPREFIX:
                 raise BadArgumentError('Invalid metadataPrefix value. It can be only used with ListRecords verb')
             record = CatalogRecord.objects.get(identifier__exact=identifier)
+            if record.state == 'published':
+                pass
+            else:
+                raise IdDoesNotExistError("No record with identifier %s is available." % identifier)
```

Comment:
should there be only record.state != 'published' with raise error or is there plan to add something in place of pass?